modules/top-level/nixpkgs.nix

```diff
@@ -0,0 +1,78 @@
+{
+  config,
+  options,
+  lib,
+  ...
+}:
+let
+  cfg = config.nixpkgs;
+  opt = options.nixpkgs;
+in
+{
+  options.nixpkgs = {
+    pkgs = lib.mkOption {
```

Comment:
What other possible options were you planning for with this? I was just curious in the example about the ux of wanting to configure nixvim's `pkgs` and it felt off to be nested under a nixpkgs attribute, for some reason. I don't know if it was just because it makes me think it implies we might allow setting a different `pkgs` for another input or something..

Reply:
As I wrote it out, though, I think the nesting makes sense.

Reply:
I intend to add most options available in nixos/hm/nix-darwin's `nixpkgs.*` modules.

In particular:
- `source`: the nixpkgs source to be imported
  (hopefully we can default to our `inputs.nixpkgs`)
- `config`: the so-called [global configuration](https://nixos.org/manual/nixpkgs/unstable/#chap-packageconfig) ([opts](https://nixos.org/manual/nixpkgs/unstable/#sec-config-options-reference))
- `overlays`: overlays to extend pkgs with
- platform options, `hostPlatform`, `buildPlatform` etc

Reply:
> I intend to add most options available in nixos/hm/nix-darwin's `nixpkgs.*` modules.
> 
> In particular:
> 
>     * `source`: the nixpkgs source to be imported
>       (hopefully we can default to our `inputs.nixpkgs`)
> 
>     * `config`: the so-called [global configuration](https://nixos.org/manual/nixpkgs/unstable/#chap-packageconfig) ([opts](https://nixos.org/manual/nixpkgs/unstable/#sec-config-options-reference))
> 
>     * `overlays`: overlays to extend pkgs with
> 
>     * platform options, `hostPlatform`, `buildPlatform` etc
> 
> 
> In contrast to the `pkgs` option added in this PR, most of these options relate to constructing a pkgs instance internally.

Awesome, that's helpful context. Thanks!